lib/Sema/CSSimplify.cpp

```diff
@@ -3058,7 +3058,8 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
       // If both have a global actor, match them.
       TypeMatchOptions subflags = getDefaultDecompositionOptions(flags);
       auto result = matchTypes(func1->getGlobalActor(), func2->getGlobalActor(),
-                               ConstraintKind::Equal, subflags, locator);
+                               ConstraintKind::Equal,
+                               subflags | TMF_ApplyingFix, locator);
```

Comment:
This is not technically correct because we not not matching to apply a fix in this case.

Reply:
Also it seems to be that the actual function type is not important here at all, a better failure would to something like - `cannot pass value isolated to <GlobalActorA> to parameter that expects a value isolated to <GlobalActorB>`

Reply:
> Also it seems to be that the actual function type is not important here at all, a better failure would to something like - `cannot pass value isolated to <GlobalActorA> to parameter that expects a value isolated to <GlobalActorB>` and `cannot convert value isolated to <GlobalActorA> to value isolated to <GlobalActorB>`

Makes sense, I'll fix as soon as possible. Thanks @xedin =]